### PR TITLE
sysbuild: Add compression options for MCUboot

### PR DESCRIPTION
**Pull request by nordicjm**
manifest: Update sdk-mcuboot
    
    Pulls in changes for image decompression support

    sysbuild: Add compression options for MCUboot
    
    Adds an option which will enable compressed image support in
    MCUboot

    tests: nrf_compress: decompression: Add mcuboot_update test
    
    Adds a test which ensures compressed update image support works
    in MCUbot

### CODE REVIEW

File: samples/nrf_compress/mcuboot_update/src/main.c
Comment (nvlsianpu) [inline samples/nrf_compress/mcuboot_update/src/main.c:1]: not a main content of this sample in this file - expect main(). rename?

File: samples/nrf_compress/mcuboot_update/src/main.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#else
+/* In "frugal" lists flags are added to response only when they evaluate to true */
+/* Note that value is evaluated twice! */
+#define ZCBOR_ENCODE_FLAG(zse, label, value)                                    \
+                (!(value) ||                                                    \
+                 (zcbor_tstr_put_lit(zse, label) && zcbor_bool_put(zse, (value))))
+#endif
+
+static struct mgmt_callback img_slot_callback;
+
```
Comment (nvlsianpu) [inline samples/nrf_compress/mcuboot_update/src/main.c:26]: comment in the code for explaining what is it doing would be helpful once someone will look at this in few months.

File: samples/nrf_compress/mcuboot_update/src/main.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+static int setup_slot_hook(void)
+{
+	img_slot_callback.callback = image_slot_state_cb;
+	img_slot_callback.event_id = MGMT_EVT_OP_IMG_MGMT_IMAGE_SLOT_STATE;
+	mgmt_callback_register(&img_slot_callback);
+
+	return 0;
+}
+
+SYS_INIT(setup_something, APPLICATION, 0);
```
Comment (nvlsianpu) [inline samples/nrf_compress/mcuboot_update/src/main.c:51]: what does this SYS_INIT?

File: tests/subsys/nrf_compress/decompression/mcuboot_update/sysbuild/mcuboot/boards/nrf54l15pdk_nrf54l15_cpuapp.conf [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+CONFIG_BOOT_MAX_IMG_SECTORS=256
+
+# Ensure that the qspi driver is disabled by default
+CONFIG_NORDIC_QSPI_NOR=n
```
Comment (nvlsianpu) [inline tests/subsys/nrf_compress/decompression/mcuboot_update/sysbuild/mcuboot/boards/nrf54l15pdk_nrf54l15_cpuapp.conf:10]:
```suggestion
CONFIG_NORDIC_SPI_NOR=n
```
nRF54l15 hasn't QSPI.

**Review by nvlsianpu — CHANGES_REQUESTED** [review]
just a few nits. Now I see why you need intel-hex output from imgtool.py

**Review by nvlsianpu — APPROVED** [review]

**Comment by nordicjm** [comment]
@carlescufi @tejlmand can you review please?

**Review by tejlmand — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**